code/modules/surgery/organs/augments_arms.dm

```diff
@@ -179,4 +179,83 @@
 		usr << "<span class='notice'>You unlock [src]'s integrated knife!</span>"
 		items_list += new /obj/item/weapon/kitchen/knife/combat/cyborg(src)
 		return 1
-	return 0
\ No newline at end of file
+	return 0
+
+
+//FISTROCKET
+
+/obj/item/organ/internal/cyberimp/fistrocket
+	name = "fist rocket implant"
+	desc = "A badass implant that lets you fire your fist off of your arm."
+	zone = "r_arm"
+	slot = "r_arm_device"
+	icon_state = "implant-toolkit"//placeholder
+	w_class = 3
+	actions_types = list(/datum/action/item_action/organ_action)
+	var/hasfist = 1
+
+/obj/item/organ/internal/cyberimp/fistrocket/ui_action_click()
+	var/mob/living/carbon/H = owner
+	var/list/hands = get_both_hands(H)
+	var/M = locate(/obj/item/rocketstump) in hands
```

Comment:
Is this check actually necessary? You've already set hasfist to 0 when the rocket hand gets fired off, and this would prevent you from using two rocket fists.